src/resources/elements/primeDesignSystem/pselect/pselect.ts

```diff
@@ -34,36 +34,33 @@ export interface IPSelectItemConfig {
 @customElement("pselect")
 export class PButton {
   @bindable disabled = false;
-  @bindable data: Array<IPSelectItemConfig>;
+  @bindable data: Array<IPSelectItemConfig> = [];
   @bindable placeholder = "Please Select...";
   @bindable searchText = "No result found.";
   @bindable searchPlaceholder ="Type to search...";
   @bindable isSearchable = false;
+  @bindable({defaultBindingMode: bindingMode.twoWay}) value?: string | string[];
 
   refSelectInput: HTMLSelectElement;
   select: SlimSelect;
   isOpen = false;
 
   attached(): void {
-    const selectConfig = {
+    this.select = new SlimSelect({
       placeholder: "<span class=\"loading\"><i class=\"fas fa-circle-notch\"></i> Loading...</span>",
       select: this.refSelectInput,
       isEnabled: !this.disabled,
       searchText: this.searchText,
       searchPlaceholder: this.searchPlaceholder,
       hideSelectedOption: true,
       showSearch: this.isSearchable,
-      data: undefined,
-    };
-
-    if (this.data) {
-      selectConfig.data = [
+      data: [
         {text: this.placeholder, placeholder: true},
         ...this.data,
-      ];
-    }
-
-    this.select = new SlimSelect(selectConfig);
+      ],
+      onChange: info => this.value = info instanceof Array ? info.map(item => item.value) : info.value,
```

Comment:
```suggestion
      onChange: info => this.value = Array.isArray(info) ? info.map(item => item.value) : info.value,
```

Nit: Maybe just preference.

Reply:
didn't know about. love it